### PR TITLE
Added RootKind and GCMethods implementation for PropertyDescriptor

### PR DESCRIPTION
**Pull request by taki-jaro**
It is needed for [servo issue 15012](https://github.com/servo/servo/issues/15012).

---
This change is [Reviewable](https://reviewable.io/reviews/servo/rust-mozjs/414)

### CODE REVIEW

File: src/rust.rs [hunk 1]
```diff
@@ -304,6 +304,11 @@ impl RootKind for Value {
     fn rootKind() -> jsapi::RootKind { jsapi::RootKind::Value }
 }
 
+impl RootKind for PropertyDescriptor {
+        #[inline(always)]
```
Comment (jdm) [inline src/rust.rs]: nit: indentation.

File: src/rust.rs [hunk 2]
```diff
@@ -836,6 +841,10 @@ impl GCMethods for Value {
     }
 }
 
+impl GCMethods for PropertyDescriptor {
+    unsafe fn initial() -> PropertyDescriptor { PropertyDescriptor::default() }
+    unsafe fn post_barrier(_ : *mut PropertyDescriptor, _ : PropertyDescriptor, _ :PropertyDescriptor) {}
+}
```
Comment (jdm) [inline src/rust.rs]: nit: add a newline after this.

File: Cargo.toml [hunk 3]
```diff
@@ -2,7 +2,7 @@
 name = "mozjs"
 description = "Rust bindings to the Mozilla SpiderMonkey JavaScript engine."
 repository = "https://github.com/servo/rust-mozjs"
-version = "0.5.0"
+version = "0.6.0"
```
Comment (jdm) [inline Cargo.toml]: This can be 0.5.1.

File: src/rust.rs (hunk 2 above)
Comment (taki-jaro) [reply src/rust.rs]: Done

**Comment by taki-jaro** [comment]
Why this stiil isn't merged 2 hours after passing tests and approval?

**Comment by jdm** [comment]
I didn't think there was any particular rush, since there's still debugging happening for the test failures, right? I've been too hasty merging other PRs like the handle one while the Servo PR isn't ready yet.

**Comment by taki-jaro** [comment]
No there is no particular rush.I was just surprised and thought there is a problem on my side.

**Comment by KiChjang** [comment]
@taki-jaro Could you rebase your commits against the master branch instead of merging? Once you've done that, you can instruct bors-servo to r=jdm.

@bors-servo delegate+

**Comment by KiChjang** [comment]
@taki-jaro No need for a new commit; you can just `@bors-servo retry` in order to re-trigger testing. Once you remove that commit, please `@bors-servo r=jdm` again.

**Comment by taki-jaro** [comment]
@KiChjang Ok, thanks

**Comment by taki-jaro** [comment]
@KiChjang It doesn't seem to work I still have failed test from 22 hours ago, the 2 retry command I gave didin't restart the test

**Comment by KiChjang** [comment]
@taki-jaro When you push a new commit, the r+ state in homu is reset, which forces you to do another r+. So if I do @bors-servo r=jdm right now, it would Just Work.

**Pull request merged**